dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -218,4 +240,295 @@ public Response deleteEnvironmentPushHistory(@Context HttpServletRequest request
+
+                this.sendErrorDeleteBundleMessage(initData, locale, e);
+            }
+        });
+
+        return Response.ok(new ResponseEntityView(
+                "Removing bundles in a separated process, the result of the operation will be notified")).build();
+    } // deleteBundlesOlderThan.
+
+    /**
+     * Deletes all failed and succeed bundles
+     * Note: the response will be notified by socket message
+     * @param request   {@link HttpServletRequest}
+     * @param response  {@link HttpServletResponse}
+     */
+    @DELETE
+    @Path("/all")
+    @Produces("application/json")
+    public Response deleteAll(@Context   final HttpServletRequest request,
+                              @Context   final HttpServletResponse response) {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        final Locale locale    = LocaleUtil.getLocale(request);
+
+        Logger.info(this, "Deleting all bundles by the user: " + initData.getUser().getUserId());
+
+        final DotSubmitter dotSubmitter = DotConcurrentFactory
+                .getInstance().getSubmitter(BUNDLE_THREAD_POOL_SUBMITTER_NAME);
+        dotSubmitter.execute(() -> {
+
+            try {
+
+                final PublishAuditStatus.Status [] statuses = Config.getCustomArrayProperty("bundle.delete.all.statuses",
+                        PublishAuditStatus.Status::valueOf, PublishAuditStatus.Status.class,
+                        ()-> new PublishAuditStatus.Status[] {FAILED_TO_SEND_TO_ALL_GROUPS, FAILED_TO_SEND_TO_SOME_GROUPS,
+                                FAILED_TO_BUNDLE, FAILED_TO_SENT, FAILED_TO_PUBLISH, SUCCESS});
+
+                final int bundleDeletedSize = this.bundleAPI.deleteAllBundles(initData.getUser(), statuses).size();
+                this.sendSuccessDeleteBundleMessage(bundleDeletedSize, initData, locale);
+            } catch (DotDataException e) {
+
+                Logger.error(this.getClass(),
+                        "Exception on deleteAll, couldn't delete bundles, exception message: " + e.getMessage(), e);
+                this.sendErrorDeleteBundleMessage(initData, locale, e);
+            }
+        });
+
+        return Response.ok(new ResponseEntityView(
+                "Removing bundles in a separated process, the result of the operation will be notified")).build();
+    } // deleteAll.
+
+    /**
+     * Deletes all failed  bundles
+     * Note: the response will be notified by socket message
+     * @param request   {@link HttpServletRequest}
+     * @param response  {@link HttpServletResponse}
+     */
+    @DELETE
+    @Path("/all/fail")
+    @Produces("application/json")
+    public Response deleteAllFail(@Context   final HttpServletRequest request,
```

Comment:
This should be a filter `DELETE /bundles?status=failed`

Reply:
it is not a filter is an argument